src/Microsoft.Dnx.Watcher.Core/DnxWatcher.cs

```diff
@@ -182,41 +181,37 @@ private void AddProjectAndDependeciesToWatcher(IProject project, IFileWatcher fi
 
             foreach (var projFile in project.ProjectDependencies)
             {
-                //var fullProjectFilePath = Path.Combine(Path.GetDirectoryName(project.ProjectFile), projFile);
-
                 AddProjectAndDependeciesToWatcher(projFile, fileWatcher);
             }
         }
 
-        private Task<string> WatchForFileChangeAsync(IFileWatcher fileWatcher, CancellationToken cancellationToken)
+        private async Task<string> WatchForFileChangeAsync(IFileWatcher fileWatcher, CancellationToken cancellationToken)
         {
-            return Task.Run(() =>
+            var tcs = new TaskCompletionSource<string>();
+
+            // Review: Do we care about the task being cancelled or just complete?
+            cancellationToken.Register(() => tcs.TrySetResult(null));
```

Comment:
For the comment: we want to task to complete. If it throws, the exception propagates and the watcher exists